CMSIS/NN/Include/arm_nnsupportfunctions.h

```diff
@@ -379,7 +385,7 @@ __STATIC_FORCEINLINE q31_t arm_nn_sat_doubling_high_mult(const q31_t m1, const q
     // as well.
     result = mult / (1UL << 31);
 
-    if ((m1 == m2) && (m1 == Q31_MIN))
+    if ((m1 == m2) && (m1 == (int32_t)Q31_MIN))
```

Comment:
Fair point, though I think it'd be better to define it as a 32bit integer in arm_math.h directly (it makes more sense to cast it to long if needed)

Reply:
@giorgio-arenarm .. True.  @christophe0606 .. just a suggestion :-)